src/components/MediaSettings/MediaSettings.vue

```diff
@@ -489,6 +489,11 @@ export default {
 	mounted() {
 		subscribe('talk:media-settings:show', this.showModal)
 		subscribe('talk:media-settings:hide', this.closeModalAndApplySettings)
+
+		const devicesPreferred = BrowserStorage.getItem('devicesPreferred')
+		if (!devicesPreferred) {
+			this.tabContent = 'devices'
+		}
```

Comment:
If I am not mistaken the `MediaSettings` is mounted throughout the life of the app, so the devices tab will be shown if there were no preferred devices during the initial load, even if they are set later. Should this be moved to `showModal` instead?

Reply:
As i understand, intention was to have it pre-opened for the very first time only (when we granting permissions).

Although as a follow-up, I'd think about forcing it in some other cases, e.g. if device used last time is not available atm, and system suggested different one, user should see the new selection

Reply:
In `showModal`, devices tab will be more forced. 

If it is in `mounted()`, If you set pref later (join a call) and change the tab, it won't force device tab next time you open media settings as it is checked only on `mounted()` hook and it doesn't get destroyed until reload. If you open media settings, change tab, and close without joining a call, it won't force it either in the next time.

However, if it is in `showModal`, it will keep enforcing devices tab when opening and closing media settings without joining a call. So the next time you open media settings, it will show devices tab.

Both behaviors are considered correct, just depending on how far we want to enforce showing devices tab.

Reply:
> As i understand, intention was to have it pre-opened for the very first time only (when we granting permissions)


No, regardless of permissions, it's for the very first time when you have no pref.

Reply:
> Although as a follow-up, I'd think about forcing it in some other cases, e.g. if device used last time is not available atm, and system suggested different one, user should see the new selection

Sounds good :+1:

> However, if it is in `showModal`, it will keep enforcing devices tab when opening and closing media settings without joining a call. So the next time you open media settings, it will show devices tab.

Right, I did not think about explicitly closing the tab without joining the call. I was thinking instead on the scenario of opening the media settings and then joining the call without closing the tab (independently of whether the selected device was fine or manually changed). Then, the next time that the user opens the media settings before joining the next call I would expect the device tab to have been automatically closed (as it would be if the user reloaded and then open the device settings before joining another call).

Having said that... now thinking about explicitly closing the tab without joining the call it might also make sense to show again the devices tab even if it was explicitly closed, as the selected device is not persisted until the call is joined. Although that may be something to change too as mentioned in the last paragraph of https://github.com/nextcloud/spreed/issues/12185

Anyway, I guess there are several scenarios to consider for a possible follow up :-)